dbt_subprojects/daily_spellbook/models/_projects/gmx/event/arbitrum/gmx_v2_arbitrum_OrderCreated.sql

```diff
@@ -0,0 +1,270 @@
+        market,
+        TRY_CAST(initial_collateral_token AS VARCHAR) AS initial_collateral_token,
+        
+        swap_path,
+        TRY_CAST(order_type AS INTEGER) AS order_type,
+        TRY_CAST(decrease_position_swap_type AS INTEGER) AS decrease_position_swap_type,
+        TRY_CAST(size_delta_usd AS DOUBLE) AS size_delta_usd,
+        TRY_CAST(initial_collateral_delta_amount AS DOUBLE) AS initial_collateral_delta_amount,
+        TRY_CAST(trigger_price AS DOUBLE) AS trigger_price,
+        TRY_CAST(acceptable_price AS DOUBLE) AS acceptable_price,
+        TRY_CAST(execution_fee AS DOUBLE) AS execution_fee,
+        TRY_CAST(callback_gas_limit AS DOUBLE) AS callback_gas_limit,
+        TRY_CAST(min_output_amount AS DOUBLE) AS min_output_amount,
+        updated_at_block,
+        TRY_CAST(updated_at_time AS DOUBLE) AS updated_at_time,
+        TRY_CAST(is_long AS BOOLEAN) AS is_long,
+        TRY_CAST(should_unwrap_native_token AS BOOLEAN) AS should_unwrap_native_token,
+        TRY_CAST(is_frozen AS BOOLEAN) AS is_frozen,
+        key
+        
+    FROM evt_data AS ED
+    LEFT JOIN evt_data_parsed AS EDP
+        ON ED.tx_hash = EDP.tx_hash
+            AND ED.index = EDP.index
+)
+
+SELECT 
+    blockchain,
+    block_time,
+    block_number,
+    tx_hash,
+    index,
+    contract_address,
+    event_name,
+    msg_sender,
+    topic1,
+    topic2,
+    
+    account,
+    receiver,
+    callback_contract,
+    ui_fee_receiver,
+    market,
+    ED.initial_collateral_token,
+    swap_path,
+    
+    CASE 
+        WHEN order_type = 0 THEN 'MarketSwap'
+        WHEN order_type = 1 THEN 'LimitSwap'
+        WHEN order_type = 2 THEN 'MarketIncrease'
+        WHEN order_type = 3 THEN 'LimitIncrease'
+        WHEN order_type = 4 THEN 'MarketDecrease'
+        WHEN order_type = 5 THEN 'LimitDecrease'
+        WHEN order_type = 6 THEN 'StopLossDecrease'
+        WHEN order_type = 7 THEN 'Liquidation'
+        ELSE NULL
+    END AS order_type,
+    CASE 
+        WHEN decrease_position_swap_type = 0 THEN 'NoSwap'
+        WHEN decrease_position_swap_type = 1 THEN 'SwapPnlTokenToCollateralToken'
+        WHEN decrease_position_swap_type = 2 THEN 'SwapCollateralTokenToPnlToken'
+        ELSE NULL
+    END AS decrease_position_swap_type,
+    
+    size_delta_usd / POWER(10, 30) AS size_delta_usd,
+    initial_collateral_delta_amount / POWER(10, collateral_token_decimals) AS initial_collateral_delta_amount,
+    trigger_price / POWER(10, 30) AS trigger_price,
+    acceptable_price / POWER(10, 30) AS acceptable_price,
+    execution_fee / POWER(10, 18) AS execution_fee,
+    callback_gas_limit,
+    min_output_amount AS min_output_amount, 
+
+    updated_at_block,
+    CASE 
+        WHEN updated_at_time = 0 THEN NULL
+        ELSE updated_at_time
+    END AS updated_at_time,
+    is_long,
+    should_unwrap_native_token,
+    is_frozen,
+    key
+
+FROM event_data AS ED
+LEFT JOIN {{ ref('gmx_v2_arbitrum_CollateralTokensData') }} AS CTD
```

Comment:
ref names will need updated to match file name changes

Reply:
Done